tests/githubUtils.test.js

```diff
@@ -343,130 +343,107 @@ describe("Github Utils Tests", () => {
+      mockErrorInput.shouldResultInPRComment = true;
+      const result = getErrorComment(mockErrorInput);
       expect(result).toBe("Test Error: Test error message");
     });
 
     test("returns null for errors that should not result in a PR comment", () => {
-      // Mock error with shouldResultInPRComment set to false
-      const mockError = {
-        message: "Test error message",
-        shouldResultInPRComment: false,
-        messagePrefix: "Test Error",
-      }
-      const result = getErrorComment(mockError);
+      mockErrorInput.shouldResultInPRComment = false;
+      const result = getErrorComment(mockErrorInput);
       expect(result).toBeNull();
     });
 
     test("returns null for errors without a shouldResultInPRComment property", () => {
-      // Mock error with a missing shouldResultInPRComment property
-      const mockError = {
-        message: "Test error message",
-        messagePrefix: "Test Error",
-      }
-      const result = getErrorComment(mockError);
-      expect(result).toBeNull();
-    });
-
-    test.each([
-      ["returns null for errors with an empty message", ""],
-      ["returns null for errors with an undefined message", undefined],
-      ["returns null for errors with a non-string message", 123],
-    ])("%s", (description, messageValue) => {
-      const mockError = {
-        message: messageValue,
-        shouldResultInPRComment: true,
-        messagePrefix: "Test Error",
-      };
-      const result = getErrorComment(mockError);
-      expect(result).toBeNull();
-    });
-
-    test.each([
-      ["returns null for errors with an empty messagePrefix", ""],
-      ["returns null for errors with an undefined messagePrefix", undefined],
-      ["returns null for errors with a non-string messagePrefix", 123],
-    ])("%s", (description, messagePrefixValue) => {
-      const mockError = {
-        message: "Test error message",
-        shouldResultInPRComment: true,
-        messagePrefix: messagePrefixValue,
-      };
-      const result = getErrorComment(mockError);
+      const result = getErrorComment(mockErrorInput);
       expect(result).toBeNull();
     });
   });
 
   describe("postPRComment", () => {
-    // Mock the Octokit createComment function
-    const createCommentMock = jest.fn();
+    const mockErrorInput = new Error("Test error message");
+    const testComment = "formatted comment string"
+    const mockGetErrorComment = jest.fn();
+    const mockCreateComment = jest.fn();
+
     const octokitMock = {
       rest: {
         issues: {
-          createComment: createCommentMock,
+          createComment: mockCreateComment,
         },
       },
     };
-    
-    const error = new Error("Test Error");
-    const testComment = "This is a test comment";
 
+    beforeAll(() => {
+      github.getOctokit.mockImplementation(() => octokitMock);
+    });
     beforeEach(() => {
-      jest.clearAllMocks();
-    })
+      mockGetErrorComment.mockClear();
```

Comment:
At last, we can use mockClear90 in these tests because each test is redefining each mock's behaviour when using mockResolve or mockReject or other resolution function. So it is ok to use mockClear in our tests